rebasehelper/output_tools/html_output_tool.py

```diff
@@ -0,0 +1,369 @@
+
+        :return: output tool name
+        """
+        return cls.NAME
+
+    @classmethod
+    def print_report_file_path(cls):
+        """Print path to the report file into the CLI"""
+        cls.colored_output('yellow', '%s report' % cls.NAME)
+        print('%s' % os.path.join(cls.html_path, 'report.' + cls.NAME))
+
+    @classmethod
+    def page_setup(cls):
+        """Generates html menu and loads css styles"""
+        global doc, tag, text
+        doc, tag, text = Doc().tagtext()
+        doc.asis('<!DOCTYPE html>')
+        with tag('html'):
+            with tag('head'):
+                with tag('meta', charset="UTF-8"):
+                    pass
+                if not cls.pure_html_output:
+                    # Generate html without css
+                    with tag('style'):
+                            text('html{width:80%;margin-left:10%;}')
+                            text('.well{max-height:500px;\
+                                    overflow-y:scroll;\
+                                    padding:10px;\
+                                    box-shadow: inset 1px 1px 5px grey, 0 0 10px grey;\
+                                    border-radius:5px;\
+                                    margin-bottom:10px;}')
+                            text('h1{text-align:center}')
+                            text('.text-center{text-align:center;}')
+
+            with tag('div'):
+                for item in ['report.html', 'dir_struct.html', 'checkers.html', 'new_logs.html', 'old_logs.html']:
+                    with tag('a', href=os.path.join(cls.html_path, item), style='margin:10px'):
+                        text(item.split('.')[0])
+
+            with tag('hr'):
+                pass
+
+    @classmethod
+    def print_results(cls, results, log, app):
+        """
+        Print results of the rebase to html
+
+        :param results results_store reference with results from the rebase
+        :param log path to a file that probably contains rebase error
+        :param app Application instance
+        """
+        if 'success' in results:
+            with tag('h1', style='color:green'):
+                text('Rebase to %s was SUCCESSFULL' % app.conf.sources)
+        else:
+            with tag('h1', style='color:red'):
+                text('Rebase to %s FAILED' % app.conf.sources)
+
+            # Print error message to the html
+            try:
+                with tag('div', klass='well'):
+                    text(results['fail'])
+
+                if log:
+                    with tag('h2'):
+                        text('Last 20 lines of %s' % log)
+
+                # Get log containing important message
+                ProcessHelper.run_subprocess(['tail', log, '-n20'], output='error.log')
+
+                with open('error.log', 'r') as f:
+                    with tag('div', klass='well'):
+
+                        for l in f.readlines():
+                            doc.asis('%s<br>' % l)
```

Comment:
I can imagine there are better ways how to present part of a log file in HTML, how about using `<pre>` tag?